core/src/test/scala/unit/kafka/raft/RaftManagerTest.scala

```diff
@@ -177,6 +201,78 @@ class RaftManagerTest {
     assertFalse(fileLocked(lockPath))
   }
 
+  @Test
+  def testMigratingZkBrokerDeletesMetadataLog(): Unit = {
+    val logDir = Some(TestUtils.tempDir().toPath)
+    val metadataDir = Some(TestUtils.tempDir().toPath)
+    val nodeId = 1
+    val config = createZkBrokerConfig(migrationEnabled = true, nodeId, logDir, metadataDir)
+    val raftManager = createRaftManager(
+      new TopicPartition("__cluster_metadata", 0),
+      config
+    )
+    raftManager.shutdown()
+
+    try {
+      KafkaRaftManager.maybeDeleteMetadataLogDir(config)
+      assertFalse(Files.exists(metadataDir.get.resolve("__cluster_metadata-0")))
+    } catch {
+      case err: Throwable => fail("Failed to delete metadata log", err)
+    }
+    assertTrue(Files.exists(metadataDir.get))
+  }
+
+  @Test
+  def testNonMigratingZkBrokerDeletesMetadataLog(): Unit = {
+    val logDir = Some(TestUtils.tempDir().toPath)
+    val metadataDir = Some(TestUtils.tempDir().toPath)
+    val nodeId = 1
+    // Use this config to create the directory
+    val config1 = createZkBrokerConfig(migrationEnabled = true, nodeId, logDir, metadataDir)
+    val raftManager = createRaftManager(
+      new TopicPartition("__cluster_metadata", 0),
+      config1
+    )
+    raftManager.shutdown()
+
+    val config2 = createZkBrokerConfig(migrationEnabled = false, nodeId, logDir, metadataDir)
+    try {
```

Comment:
ditto
```scala
    val err = assertThrows(classOf[RuntimeException], () => KafkaRaftManager.maybeDeleteMetadataLogDir(config2))
    assertEquals("Not deleting metadata log dir since migrations are not enabled.", err.getMessage)
    assertTrue(Files.exists(metadataDir.get.resolve("__cluster_metadata-0")))
```